### ISSUE TITLE
What core.async version differences should I be aware of?

### ISSUE THREAD

**Issue by devn** [LonoCloud/step.async#1]
Hello David,

Can you speak to version differences I should be aware of between the latest `core.async` and the version of it that `step.async` was built to run against?

The reason I ask is: I noticed `lonocloud.step.async/chan` currently takes `x` and passes it to `chan-named`. The arities in `chan-named` and `make-channel` by extension don't seem to support the latest arities of `core.async/chan` that allow you to pass an xform (transducer) and exception handler.

Long story short, some notes about the current usefulness of `step.async` for reasoning about edge-release `core.async` flows would be much appreciated.

Thanks!
Devin

**Comment by devn** [comment]
I saw this portion of your [talk at StrangeLoop 2014 (around the thirty minute mark)](https://www.youtube.com/watch?v=oPv-TY_XQ60#t=1739) and realize you're not really using it as a debugger, but it still remained unclear to me what "tiny slice" you actually do rely on in production.

Any chance of a completely contrived code snippet of how you're using it with Amazon's simple workflow?

(inc #'user/thanks-count),
Devin

**Comment by david-mcneil** [comment]
The main difference that I am aware of is the item you mentioned: the introduction of transducers.

The way we are using the library is roughly as:
- create a step machine representing the logic
- feed in the next external event
- wait for the machine to quiesce
- look at what is in the output channels

The logic that is in the async functions is generally just the basic core.async calls (like >!, <!) as opposed to the more exotic operators (like mult, pub, sub, etc.)

I hope that helps.
-David
